lib/remote_persistent_term/fetcher/http.ex

```diff
@@ -3,20 +3,39 @@ defmodule RemotePersistentTerm.Fetcher.Http do
   A Fetcher implementation for HTTP.
   """
   require Logger
+  alias RemotePersistentTerm.Fetcher.Http.Cache
 
   @behaviour RemotePersistentTerm.Fetcher
 
   @type t :: %__MODULE__{
-          url: String.t()
+          url: String.t(),
+          http_cache?: boolean(),
+          min_refresh_interval: pos_integer()
         }
 
-  defstruct [:url]
+  defstruct [:url, :http_cache?, :min_refresh_interval]
 
   @opts_schema [
     url: [
       type: :string,
       required: true,
       doc: "The url from which the remote term is downloaded."
+    ],
+    http_cache?: [
+      type: :boolean,
+      default: false,
+      doc: """
+      If true, the HTTP Caching spec will be used to schedule the next download.
+
+      Should avoid setting both this value to true and `refresh_interval` to a value.
+      """
+    ],
+    min_refresh_interval: [
+      type: :pos_integer,
+      default: :timer.minutes(5),
+      doc: """
+      The default refresh interval in milliseconds. This value is used when the `http_cache?` is set to true.
+      """
```

Comment:
I thought we agreed we *don't* need this and will just do what the headers tell us to do.

Reply:
The description isn't accurate actually, it is the minimum value. I'll update that. if max-age is set to a very low number, it is often set to 0, our client would then be constantly sending requests and updating our term in a continuous loop.

Reply:
ah ok, yeah the doc did confuse me, but I suppose the name does make sense. Since it is only used with http_cache, let's nest it so that http_cache would be enabled and configured like:

```elixir
[http_cache: [enabled?: true, min_refresh_interval_ms: :timer.seconds(30)]]
```

I also like to add the time unit as a suffix so its immediately clear without checking the docs.